src/test/clojure/xtdb/sql_test.clj

```diff
@@ -530,6 +530,60 @@
     "foo.a NOT LIKE_REGEX foo.b" '(not (like-regex x1 x2 ""))
     "foo.a NOT LIKE_REGEX foo.b FLAG 'i'" '(not (like-regex x1 x2 "i"))))
 
+(t/deftest test-like-regex-query-case-insensitive
```

Comment:
Was not previously working until the change to `expression.clj` - looking at the prior codegen, I believe it was essentially treating the compiled pattern (`(Pattern/compile re-literal flag-int)`) as a regular regex without any flags set. Compiling the pattern at the point of code execution seemed to fix these (and also some of the tests within `pg-regex-query`, that were failing for the same reason)